dumper/app/src/main/java/com/google/edwmigration/dumper/application/dumper/connector/redshift/RedshiftClusterUsageMetricsTask.java

```diff
@@ -0,0 +1,213 @@
+import com.google.edwmigration.dumper.application.dumper.connector.redshift.RedshiftClusterUsageMetricsTask.MetricDataPoint;
+import com.google.edwmigration.dumper.application.dumper.handle.Handle;
+import com.google.edwmigration.dumper.application.dumper.task.TaskRunContext;
+import com.google.edwmigration.dumper.plugin.lib.dumper.spi.RedshiftRawLogsDumpFormat;
+import java.io.IOException;
+import java.time.Duration;
+import java.time.ZoneOffset;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.Date;
+import java.util.List;
+import java.util.SortedMap;
+import java.util.TreeMap;
+import java.util.stream.Stream;
+import javax.annotation.Nonnull;
+
+/** Extraction task to get Redshift time series metrics from AWS CloudWatch API. */
+public class RedshiftClusterUsageMetricsTask extends AbstractAwsApiTask {
+
+  protected static enum MetricName {
+    CPUUtilization,
+    PercentageDiskSpaceUsed
+  }
+
+  protected static enum MetricType {
+    Average
+  }
+
+  @AutoValue
+  protected abstract static class MetricConfig {
+
+    public abstract MetricName name();
+
+    public abstract MetricType type();
+
+    public static MetricConfig create(MetricName name, MetricType type) {
+      return new AutoValue_RedshiftClusterUsageMetricsTask_MetricConfig(name, type);
+    }
+  }
+
+  @AutoValue
+  protected abstract static class MetricDataPoint {
+
+    public abstract Date date();
+
+    public abstract Double value();
+
+    public static MetricDataPoint create(Date date, Double value) {
+      return new AutoValue_RedshiftClusterUsageMetricsTask_MetricDataPoint(date, value);
+    }
+  }
+
+  private static final String REDSHIFT_NAMESPACE = "AWS/Redshift";
+  private static final DateTimeFormatter DATE_FORMAT =
+      DateTimeFormatter.ofPattern("yyyy-MM-dd HH/mm/ss").withZone(ZoneOffset.UTC);
+
+  private final ZonedDateTime currentTime;
+  private final ZonedInterval interval;
+  private final List<MetricConfig> metrics;
+
+  public RedshiftClusterUsageMetricsTask(
+      AWSCredentialsProvider credentialsProvider,
+      ZonedDateTime currentTime,
+      ZonedInterval interval,
+      String zipEntryName,
+      ImmutableList<MetricConfig> metrics) {
+    super(
+        credentialsProvider,
+        zipEntryName,
+        RedshiftRawLogsDumpFormat.ClusterUsageMetrics.Header.class);
+    this.interval = interval;
+    this.metrics = metrics;
+    this.currentTime = currentTime;
+  }
+
+  @Override
+  protected Void doRun(TaskRunContext context, @Nonnull ByteSink sink, Handle handle)
+      throws IOException {
+    writeRecordsCsv(
+        sink,
+        listClusters().stream()
+            .flatMap(
+                cluster ->
+                    getClusterMetrics(cluster.getClusterIdentifier()).entrySet().stream()
+                        .map(
+                            metricEntry ->
+                                Stream.of(
+                                        new Object[] {
```

Comment:
I would suggest wrapping `Object[]` into a wrapper (e.g. `CsvRow`) and then adding the operation
`CsvRow prepend(Object... values)` that would encapsulate this expression `Stream.of(...)...toArray()`.